tests/test_ptc.py

```diff
@@ -88,6 +90,62 @@ def setUp(self):
+        localDataset.badAmps = [localDataset.ampNames[0], localDataset.ampNames[1]]
+
+        for ampName in localDataset.ampNames:
+            localDataset.inputExpIdPairs[ampName] = np.repeat(1, nSignalPoints)
+            localDataset.expIdMask[ampName] = [True, False, True, True, False, True, False, True, True,
+                                               True, True, False, True, False, True]
+            localDataset.expIdMask[ampName] = np.arange(nSignalPoints)
+            # overwrite rawExpTimes from setup
+            localDataset.rawExpTimes[ampName] = np.arange(nSignalPoints)
+            localDataset.rawMeans[ampName] = self.flux*np.arange(nSignalPoints)
+            localDataset.rawVars[ampName] = self.noiseSq + self.c1*self.flux*np.arange(nSignalPoints)
+            localDataset.photoCharge[ampName] = np.repeat(np.nan, nSignalPoints)
+            localDataset.gain[ampName] = self.gain
+            localDataset.gainErr[ampName] = 0.1
+            localDataset.noise[ampName] = self.noiseSq
+            localDataset.noiseErr[ampName] = 2.0
+            localDataset.ptcFitPars[ampName] = np.array([10.0, 1.5, 1e-6])
+            localDataset.ptcFitParsError[ampName] = np.array([1.0, 0.2, 1e-7])
+            localDataset.ptcFitChiSq[ampName] = 1.0
+            localDataset.covariances[ampName] = np.full((nSignalPoints, nSideCovMatrix, nSideCovMatrix),
+                                                        105.0)
+            localDataset.covariancesModel[ampName] = np.full((nSignalPoints, nSideCovMatrix, nSideCovMatrix),
+                                                             100.0)
+            localDataset.covariancesSqrtWeights[ampName] = np.full((nSignalPoints, nSideCovMatrix,
+                                                                   nSideCovMatrix), 10.0)
+            localDataset.aMatrix[ampName] = np.full((nSideCovMatrix, nSideCovMatrix), 1e-6)
+            localDataset.bMatrix[ampName] = np.full((nSideCovMatrix, nSideCovMatrix), np.nan)
+            localDataset.covariancesNoB[ampName] = np.full((nSignalPoints, nSideCovMatrix, nSideCovMatrix),
+                                                           np.nan)
+            localDataset.covariancesModelNoB[ampName] = np.full((nSignalPoints, nSideCovMatrix,
+                                                                nSideCovMatrix), np.nan)
+            localDataset.covariancesSqrtWeightsNoB[ampName] = np.full((nSignalPoints, nSideCovMatrix,
+                                                                      nSideCovMatrix), np.nan)
+            localDataset.aMatrixNoB[ampName] = np.full((nSideCovMatrix, nSideCovMatrix), np.nan)
+            localDataset.finalVars[ampName] = self.noiseSq + self.c1*self.flux*np.arange(nSignalPoints)
+            localDataset.finalModelVars[ampName] = np.repeat(100.0, nSignalPoints)
+            localDataset.finalMeans[ampName] = self.flux*np.arange(nSignalPoints)
+
+        # Round trip (write-read) with FITS
```

Comment:
This should test equivalence of the local dataset and the one read back from disk.  `tempfile` will also handle the file IO so you don't need to.

Reply:
Yes, you want to write it to fits and yaml and then read both back in and ensure they agree with what you had originally (and yes this test should be with the calibration class code in ip_isr.